jpo-deduplicator/src/main/java/us/dot/its/jpo/deduplicator/deduplicator/processors/OdeTimJsonProcessor.java

```diff
@@ -45,6 +58,23 @@ public boolean isDuplicate(OdeTimData lastMessage, OdeTimData newMessage) {
                 return false;
             }
 
+            // Add null checks for payload and data
+            if (lastMessage == null || lastMessage.getPayload() == null || lastMessage.getPayload().getData() == null ||
+                    newMessage == null || newMessage.getPayload() == null
+                    || newMessage.getPayload().getData() == null) {
+                logger.warn("One or both TIM messages have null payload or data, treating as non-duplicate");
+                return false;
+            }
+
+            // Check if the TIM packet IDs are the same
+            TravelerInformationMessageFrame oldTim = (TravelerInformationMessageFrame) lastMessage.getPayload().getData();
+            TravelerInformationMessageFrame newTim = (TravelerInformationMessageFrame) newMessage.getPayload().getData();
+            String oldTimPacketId = oldTim.getValue().getPacketID().getValue();
+            String newTimPacketId = newTim.getValue().getPacketID().getValue();
+
+            if (!oldTimPacketId.equals(newTimPacketId)) {
+                return false;
```

Comment:
(question) did this logic change? Should we also be verifying that the msgCnt field is the same before saying that the TIM message is a duplicate?

Reply:
It seems like there really wasn't any logic at all for TIM deduplication besides time. I can add that.

Reply:
I seem some logic in the topology that relates to how it's doing the keying. So maybe that's how it's done. @John-Wiens do you have an opinion on this?

Reply:
Aw, yeah I guess the keying does that so my additional logic is probably redundant.

Reply:
I have gone ahead and removed this redundant logic since the key does indeed do this.